lib/command/run.rb

```diff
@@ -184,13 +182,22 @@ def create_runner_workload # rubocop:disable Metrics/MethodLength
       end
     end
 
-    def update_runner_workload # rubocop:disable Metrics/MethodLength
-      step("Updating runner workload '#{runner_workload}'") do
-        @expected_deployed_version = cp.cron_workload_deployed_version(runner_workload)
-        should_update = false
+    def update_runner_workload # rubocop:disable Metrics/CyclomaticComplexity, Metrics/MethodLength, Metrics/PerceivedComplexity
+      should_update = false
+      spec = nil
 
+      step("Checking if runner workload '#{runner_workload}' needs to be updated") do
+        _, original_container_spec = base_workload_specs(original_workload)
         spec, container_spec = base_workload_specs(runner_workload)
 
+        # Keep env synced between original workload and runner workload
+        original_env_str = original_container_spec["env"]&.sort_by { |env| env["name"] }.to_s
+        env_str = container_spec["env"]&.sort_by { |env| env["name"] }.to_s
```

Comment:
We sort by name and convert to string to make it easier to compare if the env values are different.